File: Patterns/Hexagon/hexagon.java
public class HexagonPattern {
    public static void main(String[] args) {
        int n = 4;       
        int width = 13;  

        for (int i = 0; i < n; i++) {
            int stars = width/2 + i*2;
            int spaces = (width - stars)/2;
            for (int s = 0; s < spaces; s++) System.out.print(" ");
            for (int s = 0; s < stars; s++) System.out.print("*");
            System.out.println();
        }

        for (int i = n-2; i >= 0; i--) {
            int stars = width/2 + i*2;
            int spaces = (width - stars)/2;
            for (int s = 0; s < spaces; s++) System.out.print(" ");
            for (int s = 0; s < stars; s++) System.out.print("*");
            System.out.println();
        }
    }
}

File: Patterns/Hexagon/hexagon.py
n = 4  
width = 13  
for i in range(n):
    stars = width//2 + i*2
    spaces = (width - stars)//2
    print(' '*spaces + '*'*stars)
for i in range(n-2, -1, -1):
    stars = width//2 + i*2
    spaces = (width - stars)//2
    print(' '*spaces + '*'*stars)
